### PR TITLE
011 view job family capability

### PR DESCRIPTION
**Pull request by malachi-mccarney**

### CODE REVIEW

**Review by martin-doherty1 — COMMENTED** [review]
looks good

**Review by ccurran01 — COMMENTED** [review]
@malachi-mccarney Could we merge main into this please? Should make it a bit easier to review

**Comment by malachi-mccarney** [comment]
> @malachi-mccarney Could we merge main into this please? Should make it a bit easier to review

It wont let me commit or push up as it seems to be already merged with main

File: Errors/FailedToGetJobFamilies.ts [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export class FailedToGetJobFamiliesError implements Error {
```
Comment (ccurran01) [inline Errors/FailedToGetJobFamilies.ts:1]: I think this should be extending error instead 

```suggestion
export class FailedToGetJobFamiliesError extends Error {
```

File: model/JobFamily.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export interface JobFamily
```
Comment (ccurran01) [inline model/JobFamily.ts:1]: Could we rename this to be JobFamilyResponse. We should update the file name to match
```suggestion
export interface JobFamilyResponse
```

File: service/JobFamilyService.ts [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+        const response:AxiosResponse = await axios.get(`${baseURL}/api/job-family`)
+
+        return response.data
+    } catch(e)
+    {
+        throw new FailedToGetJobFamiliesError('Could not get job families')
+    }
+}
+
+
```
Comment (ccurran01) [inline service/JobFamilyService.ts:20]: Do we need this blank space?

File: Errors/FailedToGetJobFamilies.ts [hunk 4]
```diff
@@ -0,0 +1,8 @@
+export class FailedToGetJobFamiliesError extends Error {
+    constructor(message: string) {
+        super();
+        this.message = message;
+    }
+    message: string;
+    name: string;
+}
```
Comment (ccurran01) [inline Errors/FailedToGetJobFamilies.ts:8]:
```suggestion
export class FailedToGetJobFamiliesError extends Error {
    constructor(message: string) {
        super(message);
    }
}
```

File: test/unit/service/JobFamilyServiceTest.ts [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+        mock.onGet(jobFamilyService.URL).reply(200, data);
+
+        let results = await jobFamilyService.getJobFamilies();
+
+        expect(results[0]).to.deep.equal(jobFamily);
+    })
+
+    it('should throw exception when 500 error returned from axios', async () => {
+        var mock = new MockAdapter(axios);
```
Comment (ccurran01) [inline test/unit/service/JobFamilyServiceTest.ts:31]: could we swap var out to const here please 

```suggestion
        const mock = new MockAdapter(axios);
```

File: test/unit/service/JobFamilyServiceTest.ts [hunk 6]
```diff
@@ -0,0 +1,42 @@
+import { JobFamilyResponse } from "../../../model/JobFamilyResponse";
+import axios from "axios"
+import MockAdapter from "axios-mock-adapter"
+
+var chai = require('chai');
```
Comment (ccurran01) [inline test/unit/service/JobFamilyServiceTest.ts:5]:
```suggestion
const chai = require('chai');  
```

File: test/unit/service/JobFamilyServiceTest.ts [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+        expect(results[0]).to.deep.equal(jobFamily);
+    })
+
+    it('should throw exception when 500 error returned from axios', async () => {
+        var mock = new MockAdapter(axios);
+
+        mock.onGet(jobFamilyService.URL).reply(500);
+
+        try {
+          var error = await jobFamilyService.getJobFamilies();
```
Comment (ccurran01) [inline test/unit/service/JobFamilyServiceTest.ts:36]: Not sure that this variable is required 

```suggestion
     await jobFamilyService.getJobFamilies();
```

**Review by martin-doherty1 — APPROVED** [review]

**Review by ccurran01 — APPROVED** [review]

**Pull request merged**